### PR TITLE
refactor(core, connector): fix code to fit new ESlint rule

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

1. Separate connector route UTs to make each testing file under 300 lines.
2. Fix WeChat Web and WeChat Native connector error handler of getUserInfo() to fit the max-complexity of 5 for each code block.

## Linear Issue Reference

LOG-2405

## Testing

Passed UTs.

### CODE REVIEW

File: packages/connector-wechat-native/src/index.ts [hunk 1]
```diff
@@ -92,25 +90,27 @@ export default class WeChatNativeConnector implements SocialConnector {
...
 
-        throw new Error(errmsg);
-      }
+      // Response properties of user info can be separated into two groups: (1) {unionid, headimgurl, nickname}, (2) {errcode, errmsg}.
+      // These two groups are mutually exclusive: if group (1) is not empty, group (2) should be empty and vice versa.
+      // 'errmsg' and 'errcode' turn to be non-empty values or empty values at the same time. Hence, if 'errmsg' is non-empty then 'errcode' should be non-empty.
+
+      assert(errcode !== 40_001, new ConnectorError(ConnectorErrorCodes.SocialAccessTokenInvalid));
+      assert(!errcode, new Error(errmsg));
+      assert(openid, new Error(errmsg));
```
Comment (xiaoyijun) [inline packages/connector-wechat-native/src/index.ts:106]: Looks confusing here, maybe guard it when we first get this `openid` in line: 84.

File: packages/connector-wechat/src/index.ts [hunk 2]
```diff
@@ -92,25 +90,27 @@ export default class WeChatConnector implements SocialConnector {
...
 
-        throw new Error(errmsg);
-      }
+      // Response properties of user info can be separated into two groups: (1) {unionid, headimgurl, nickname}, (2) {errcode, errmsg}.
+      // These two groups are mutually exclusive: if group (1) is not empty, group (2) should be empty and vice versa.
+      // 'errmsg' and 'errcode' turn to be non-empty values or empty values at the same time. Hence, if 'errmsg' is non-empty then 'errcode' should be non-empty.
+
+      assert(errcode !== 40_001, new ConnectorError(ConnectorErrorCodes.SocialAccessTokenInvalid));
+      assert(!errcode, new Error(errmsg));
+      assert(openid, new Error(errmsg));
```
Comment (xiaoyijun) [inline packages/connector-wechat/src/index.ts:106]: ditto.

File: packages/connector-wechat-native/src/index.ts [hunk 3]
```diff
@@ -92,25 +90,27 @@ export default class WeChatNativeConnector implements SocialConnector {
...
+      // fact is that we still got 40001 'invalid credentials' response.
 
-        throw new Error(errmsg);
-      }
+      // Response properties of user info can be separated into two groups: (1) {unionid, headimgurl, nickname}, (2) {errcode, errmsg}.
+      // These two groups are mutually exclusive: if group (1) is not empty, group (2) should be empty and vice versa.
+      // 'errmsg' and 'errcode' turn to be non-empty values or empty values at the same time. Hence, if 'errmsg' is non-empty then 'errcode' should be non-empty.
+
+      assert(errcode !== 40_001, new ConnectorError(ConnectorErrorCodes.SocialAccessTokenInvalid));
+      assert(!errcode, new Error(errmsg));
```
[the remark this answers is not in the document]
Comment (darcyYe) [reply packages/connector-wechat-native/src/index.ts:105]: if `errcode` is 0 or undefined means that the request was successful.
image

File: packages/connector-wechat-native/src/index.ts (hunk 1 above)
Comment (darcyYe) [reply packages/connector-wechat-native/src/index.ts]: Good call, fixed.

File: packages/connector-wechat-native/src/index.ts [hunk 4]
```diff
@@ -92,25 +91,26 @@ export default class WeChatNativeConnector implements SocialConnector {
...
-        // fact is that we still got 40001 'invalid credentials' response.
-        if (errcode === 40_001) {
-          throw new ConnectorError(ConnectorErrorCodes.SocialAccessTokenInvalid);
-        }
+      // 'openid' is defined as a required input argument in WeChat API doc, but it does not necessarily to
+      // be the return value from getAccessToken per testing.
+      // In another word, 'openid' is required but the response of getUserInfo is consistent as long as
+      // access_token is valid.
+      // We are expecting to get 41009 'missing openid' response according to the developers doc, but the
+      // fact is that we still got 40001 'invalid credentials' response.
```
Comment (IceHe) [inline packages/connector-wechat-native/src/index.ts:99]: seems that this comments should be moved above 

```ts
assert(openid, new Error('`openid` is required for WeChat API.'));
```

File: packages/connector-wechat/src/index.ts [hunk 5]
```diff
@@ -92,25 +91,26 @@ export default class WeChatConnector implements SocialConnector {
...
-        // fact is that we still got 40001 'invalid credentials' response.
-        if (errcode === 40_001) {
-          throw new ConnectorError(ConnectorErrorCodes.SocialAccessTokenInvalid);
-        }
+      // 'openid' is defined as a required input argument in WeChat API doc, but it does not necessarily to
+      // be the return value from getAccessToken per testing.
+      // In another word, 'openid' is required but the response of getUserInfo is consistent as long as
+      // access_token is valid.
+      // We are expecting to get 41009 'missing openid' response according to the developers doc, but the
+      // fact is that we still got 40001 'invalid credentials' response.
```
Comment (IceHe) [inline packages/connector-wechat/src/index.ts:99]: ditto

File: packages/connector-wechat-native/src/index.test.ts [hunk 6]
```diff
@@ -106,6 +106,17 @@ describe('getUserInfo', () => {
     });
   });
 
+  it('throws error if `openid` is missing', async () => {
+    nock(userInfoEndpointUrl.origin).get(userInfoEndpointUrl.pathname).query(parameters).reply(0, {
+      unionid: 'this_is_an_arbitrary_wechat_union_id',
+      headimgurl: 'https://github.com/images/error/octocat_happy.gif',
+      nickname: 'wechat bot',
```
Comment (demonzoo) [inline packages/connector-wechat-native/src/index.test.ts:113]: why don't these param names use camelCase?

Comment (darcyYe) [reply packages/connector-wechat-native/src/index.test.ts:113]: Followed the type definition of WeChat API response.

Comment (demonzoo) [reply packages/connector-wechat-native/src/index.test.ts:113]: ok

File: packages/connector-wechat-native/src/index.ts [hunk 7]
```diff
@@ -79,10 +79,9 @@ export default class WeChatNativeConnector implements SocialConnector {
     return { accessToken, openid };
   };
 
-  // FIXME:
-  // eslint-disable-next-line complexity
   public getUserInfo: GetUserInfo = async (accessTokenObject) => {
     const { accessToken, openid } = accessTokenObject;
+    assert(openid, new Error('`openid` is required for WeChat API.'));
```
Comment (demonzoo) [inline packages/connector-wechat-native/src/index.ts:84]: nit: required `by`

File: packages/connector-wechat-native/src/index.ts [hunk 8]
```diff
@@ -92,25 +91,26 @@ export default class WeChatNativeConnector implements SocialConnector {
...
-        // 'openid' is defined as a required input argument in WeChat API doc, but it does not necessarily to
-        // be the return value from getAccessToken per testing.
-        // In another word, 'openid' is required but the response of getUserInfo is consistent as long as
-        // access_token is valid.
-        // We are expecting to get 41009 'missing openid' response according to the developers doc, but the
-        // fact is that we still got 40001 'invalid credentials' response.
-        if (errcode === 40_001) {
-          throw new ConnectorError(ConnectorErrorCodes.SocialAccessTokenInvalid);
-        }
+      // 'openid' is defined as a required input argument in WeChat API doc, but it does not necessarily to
```
Comment (demonzoo) [inline packages/connector-wechat-native/src/index.ts:94]: nit: does not necessarily `have` to be

File: packages/connector-wechat-native/src/index.ts [hunk 9]
```diff
@@ -92,25 +91,26 @@ export default class WeChatNativeConnector implements SocialConnector {
...
+      // be the return value from getAccessToken per testing.
+      // In another word, 'openid' is required but the response of getUserInfo is consistent as long as
```
Comment (demonzoo) [inline packages/connector-wechat-native/src/index.ts:96]: nit: `In other words`

File: packages/connector-wechat-native/src/index.ts [hunk 10]
```diff
@@ -92,25 +91,26 @@ export default class WeChatNativeConnector implements SocialConnector {
...
+      // access_token is valid.
+      // We are expecting to get 41009 'missing openid' response according to the developers doc, but the
+      // fact is that we still got 40001 'invalid credentials' response.
 
-        throw new Error(errmsg);
-      }
+      // Response properties of user info can be separated into two groups: (1) {unionid, headimgurl, nickname}, (2) {errcode, errmsg}.
+      // These two groups are mutually exclusive: if group (1) is not empty, group (2) should be empty and vice versa.
+      // 'errmsg' and 'errcode' turn to be non-empty values or empty values at the same time. Hence, if 'errmsg' is non-empty then 'errcode' should be non-empty.
```
Comment (demonzoo) [inline packages/connector-wechat-native/src/index.ts:103]: nit: turn to ~~be~~ non-empty values

File: packages/connector-wechat/src/index.ts [hunk 11]
```diff
@@ -79,10 +79,9 @@ export default class WeChatConnector implements SocialConnector {
     return { accessToken, openid };
   };
 
-  // FIXME:
-  // eslint-disable-next-line complexity
   public getUserInfo: GetUserInfo = async (accessTokenObject) => {
     const { accessToken, openid } = accessTokenObject;
+    assert(openid, new Error('`openid` is required for WeChat API.'));
```
Comment (demonzoo) [inline packages/connector-wechat/src/index.ts:84]: nit: required `by` WeChat API

**Review by demonzoo — APPROVED** [review]

**Pull request merged**